torchvision/datasets/folder.py

```diff
@@ -32,18 +32,25 @@ def is_image_file(filename):
     return has_file_allowed_extension(filename, IMG_EXTENSIONS)
 
 
-def make_dataset(dir, class_to_idx, extensions):
+def make_dataset(dir, class_to_idx, extensions=None, is_valid_file=None):
     images = []
     dir = os.path.expanduser(dir)
+    if extensions is None and is_valid_file is None:
+        raise ValueError("Both extensions and is_valid_file cannot be None")
+    if extensions is not None and is_valid_file is not None:
+        raise ValueError("One of the extensions and is_valid_file should be None")
+
     for target in sorted(class_to_idx.keys()):
         d = os.path.join(dir, target)
         if not os.path.isdir(d):
             continue
-
+        if extensions is not None:
+            def is_valid_file(x):
```

Comment:
One more minor nit: this can be moved outside of the for loop. This will avoid having to recreate the same function over and over again, for each class. Can you move it outside of the loop?

Reply:
Ya sure